doc/development/tutorials/extending_syntax.rst

```diff
@@ -0,0 +1,224 @@
+.. _tutorial-extending-syntax:
+
+Extending syntax with roles and directives
+==========================================
+
+Overview
+--------
+
+The syntax of both reStructuredText and MyST can be extended
+by creating new directives - for block-level elements -
+and roles - for inline elements.
+
+In this tutorial we shall extend Sphinx to add:
+
+* A ``hello`` role, that will simply output the text ``Hello {text}!``.
+* A ``hello`` directive, that will simply output the text ``Hello {text}!``,
+  as a paragraph.
```

Comment:
Maybe worth using distinct names for the role and directive?

Reply:
Hmm, I'd prefer to keep them the same, since that is kinda the point; that they are both doing exactly the same thing, except one is at an inline-level and the other at block-level